File: ios/DailyPlanet/WhiteLabelConfig.m
//
//  WhiteLabelConfig.m
//  DailyPlanet
//
//  Created by Davidson Marra on 10/03/23.
//

#import <Foundation/Foundation.h>
#import "WhiteLabelConfig.h"

@implementation WhiteLabelConfig

  RCT_EXPORT_MODULE(WhiteLabelConfig);
 
  RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getAppName) {
    return [[NSBundle mainBundle] objectForInfoDictionaryKey:@"CFBundleDisplayName"];
  }

  - (NSDictionary *)constantsToExport {
    return @{
      @"theme": @{
        @"background": @"#f3f8fe",
        @"primary": @"#8db5d7",
        @"secondary": @"#fee344",
        @"title": @"#1c1c1c",
        @"heroTitle": @"#f1f1f1",
        @"brightness": @"rgba(0, 0, 0, 0.7)",
        @"cardBackground": @"#1c1c1c33",
        @"mode": @"dark"
      }
    };
  }

  + (BOOL)requiresMainQueueSetup {
    return YES;
  }

@end

File: android/app/src/dailyPlanet/java/com/whitelabelnews/WhiteLabelConfig.java
// WhiteLabelConfig.java for WL1 product flavor

package com.whitelabelnews;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

import javax.annotation.Nonnull;
import java.util.Map;
import java.util.HashMap;

public class WhiteLabelConfig extends ReactContextBaseJavaModule {

    @Nonnull
    private final String appName;

    public WhiteLabelConfig(@Nonnull ReactApplicationContext reactContext) {
        super(reactContext);

        this.appName = reactContext
                .getApplicationContext()
                .getResources()
                .getString(R.string.white_label_app_name);
    }


    @Nonnull
    @Override
    public String getName() {
        return "WhiteLabelConfig";
    }

    @Nonnull
    @ReactMethod(isBlockingSynchronousMethod = true)
    public String getAppName() {
        return appName;
    }

    @Nonnull
    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();

        WritableMap themeMap = Arguments.createMap();
        themeMap.putString("background", "#f3f8fe");
        themeMap.putString("primary", "#8db5d7");
        themeMap.putString("secondary", "#fee344");
        themeMap.putString("title", "#1c1c1c");
        themeMap.putString("heroTitle", "#f1f1f1");
        themeMap.putString("brightness", "rgba(0, 0, 0, 0.7)");
        themeMap.putString("cardBackground", "#1c1c1c33");

        constants.put("theme", themeMap);
        constants.putString("searchHero", "superman");

        return constants;
    }
}